.pre-commit-config.yaml

```diff
@@ -0,0 +1,17 @@
+# https://pre-commit.com
+
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.0.1
+    hooks:
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+      - id: check-yaml
+      - id: debug-statements
+      - id: mixed-line-ending
+# black as python formatter
+  - repo: https://github.com/psf/black
+    rev: 22.3.0
```

Comment:
Is there a way that the run of black could use the ctsm_py conda environment? As well as the run of black to use the python/pyproject.toml file? That would keep the version and options for black consistent across all uses of it. the line-length is especially the thing that we want to make sure is consistent across all uses of black as well as consistent with the line length for pylint.